db/sql/sql_test.go

```diff
@@ -411,6 +412,32 @@ var _ = Describe("Sql", func() {
 				Expect(resSql).To(Equal(expectedSql))
 				Expect(param).To(Equal(expectedParam))
 			})
+			It("should create like query with one parameter", func() {
+				search_value := search.NewSearchField("123")
+				filter := map[string]interface{}{"test_string": search_value}
+
+				res, err := AddFilterToQuery(testSchema, query, filter, false)
+
+				Expect(err).ToNot(HaveOccurred())
+				resSql, param, err := res.ToSql() 
+				Expect(err).ToNot(HaveOccurred())
+				expectedQuery = expectedQuery.Where(squirrel.Like{"`test_string`": "%123%"})
+				expectedSql, expectedParam, _ := expectedQuery.ToSql()
+				Expect(resSql).To(Equal(expectedSql))
+				Expect(param).To(Equal(expectedParam))
+			})
+			It("should create like query with two parameter", func() {
+				string_search := search.NewSearchField("123")
+				number_search := search.NewSearchField("42")
+				filter := map[string]interface{}{"test_string": string_search, "test_number": number_search}
+
+				res, err := AddFilterToQuery(testSchema, query, filter, false)
+
+				Expect(err).ToNot(HaveOccurred())
+				resSql, param, err := res.ToSql()
+				Expect(err).ToNot(HaveOccurred())
+				checkAndStatement(resSql, "`test_string` LIKE ?", "`test_number` LIKE ?", param, []interface{}{ "%123%","%42%"})
+			})
```

Comment:
Let's add an unit test of SQL statement mixed with substr search enabled/disabled fields.